### PR TITLE
Importer improvements

### PR DESCRIPTION
**Pull request by SferaDev**

### CODE REVIEW

File: packages/importer/src/utils.ts [hunk 1]
```diff
@@ -58,36 +73,64 @@ export function guessType(value: string) {
...
-  } else if ((a === 'float' && b === 'int') || (a === 'int' && b === 'float')) {
+export function castType(schemaType: string, valueType: string) {
+  if (schemaType === valueType) {
+    return schemaType;
+  } else if ((schemaType === 'float' && valueType === 'int') || (schemaType === 'int' && valueType === 'float')) {
     return 'float';
-  } else if (a === 'text' || b === 'text') {
+  } else if (schemaType === 'text' || valueType === 'text') {
     return 'text';
+  } else if (schemaType === 'link') {
```
Comment (SferaDev) [inline packages/importer/src/utils.ts:87]:
```suggestion
  } else if (schemaType === 'link' && valueType === 'string') {
```

File: packages/importer/src/utils.ts [hunk 2]
```diff
@@ -58,36 +73,64 @@ export function guessType(value: string) {
...
+    return 'link';
+  } else if (schemaType === 'datetime') {
```
Comment (SferaDev) [inline packages/importer/src/utils.ts:89]:
```suggestion
  } else if (schemaType === 'datetime' && valueType === 'string') {
```

File: packages/importer/src/processor.test.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import { Schemas, XataApiClient } from '@xata.io/client';
 import { describe, expect, test, vi } from 'vitest';
-import { compareSquema, createProcessor, TableInfo } from './processor';
+import { compareSchema, createProcessor, TableInfo } from './processor';
```
Comment (gimenete) [inline packages/importer/src/processor.test.ts:3]: Nice catch! 😄

File: packages/importer/src/utils.ts [hunk 4]
```diff
@@ -16,7 +16,18 @@ export function parseArray(value: string) {
...
-export function guessTypes(lines: string[][], columns: string[]): string[] {
+function parseStringArray(value: string) {
+  return value
+    .split(',')
+    .map((s) => s.trim())
+    .filter((s) => s.length > 0);
+}
+
+export function parseArray(value: string): string[] | null {
+  return parseJSONArray(value) ?? parseStringArray(value);
```
Comment (gimenete) [inline packages/importer/src/utils.ts:27]: ✨

File: packages/importer/src/utils.ts [hunk 5]
```diff
@@ -58,36 +73,64 @@ export function guessType(value: string) {
...
+    if (nullValues.includes(val)) {
+      return null;
+    } else if (type === 'int') {
       return Number.isSafeInteger(num) && val !== '' ? num : null;
     } else if (type === 'float') {
-      const num = +val;
       return Number.isFinite(num) && val !== '' ? num : null;
     } else if (type === 'bool') {
-      return ['true', 'false'].includes(val) ? val === 'true' : null;
+      if (val === 'true' || num === 1) {
```
Comment (gimenete) [inline packages/importer/src/utils.ts:116]: LGTM. Maybe in the future we make this configurable to be more strict, or we allow the user to specify which values should be casted to `true`.

File: packages/importer/src/utils.ts [hunk 6]
```diff
@@ -58,36 +73,64 @@ export function guessType(value: string) {
...
+      }
     } else if (type === 'multiple') {
       return parseArray(val);
     } else if (type === 'email') {
       return val || null;
+    } else if (type === 'link') {
+      return val ? String(val) : null;
+    } else if (type === 'datetime') {
+      const date = new Date(val);
+      return !isNaN(date.getTime()) ? date : null;
```
Comment (gimenete) [inline packages/importer/src/utils.ts:131]:
```suggestion
      return isNaN(date.getTime()) ? null : date;
```

Nitpick. You can completely ignore this suggesion 😄

**Review by gimenete — APPROVED** [review]
💯

**Pull request merged**